test/kubernetes/e2e/features/route_options/suite.go

```
@@ -32,55 +40,213 @@ func NewTestingSuite(ctx context.Context, testInst *e2e.TestInstallation) suite.
-		},
+		s.getterForMeta(&basicRtoTargetRefMeta),
 		core.Status_Accepted,
 		defaults.KubeGatewayReporter,
 	)
 }
 
 func (s *testingSuite) TestConfigureRouteOptionsWithFilterExtension() {
-	s.T().Cleanup(func() {
-		err := s.testInstallation.Actions.Kubectl().DeleteFile(s.ctx, filterExtensioManifest)
-		s.NoError(err, "can delete manifest")
-		s.testInstallation.Assertions.EventuallyObjectsNotExist(s.ctx, proxyService, proxyDeployment)
-	})
+	s.assertEventuallyCurlRespondsWith(expectedResponseWithBasicHeader)
+
+	// TODO(npolshak): Statuses are not supported for filter extensions yet
+}
+
+func (s *testingSuite) TestConfigureInvalidRouteOptionsWithTargetRef() {
+	// Check status is rejected on RouteOption
+	s.testInstallation.Assertions.EventuallyResourceStatusMatchesState(
+		s.getterForMeta(&badRtoTargetRefMeta),
+		core.Status_Rejected,
+		defaults.KubeGatewayReporter,
+	)
+}
 
-	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, filterExtensioManifest)
-	s.NoError(err, "can apply targetRefManifest")
+// TODO(jbohanon) statuses not implemented for extension ref
+// func (s *testingSuite) TestConfigureInvalidRouteOptionsWithFilterExtension() {
+// 	// Check status is rejected on RouteOption
+// 	s.testInstallation.Assertions.EventuallyResourceStatusMatchesState(
+// 		s.getterForMeta(&badRtoMeta),
+// 		core.Status_Rejected,
+// 		defaults.KubeGatewayReporter,
+// 	)
+// }
 
-	s.testInstallation.Assertions.EventuallyObjectsExist(s.ctx, proxyService, proxyDeployment)
+// will fail until manual setup added
+func (s *testingSuite) TestConfigureRouteOptionsWithMultipleTargetRefManualSetup() {
+	// Manually apply our manifests so we can assert that basic rto exists before applying extra rto.
+	// This is needed because our solo-kit clients currently do not return creationTimestamp
+	err := s.testInstallation.Actions.Kubectl().ApplyFile(s.ctx, setupManifest)
+	s.NoError(err, "can apply "+setupManifest)
```

I think the `EventuallyObjects` exist check is still valid after we apply the `test/kubernetes/e2e/features/route_options/testdata/setup.yaml`. We don't need it for any of the other manifests, but it would be good to assert the gateway is created by the deployer.

---

But none of the features of the route options plugin affect the behavior of the deployer, nor are they contingent upon the deployer's success. Given that, I think it makes more sense to let the routing fail and depend on the deployer tests to catch issues that may arise with that feature.

---

I think it's more similar to the post installation checks were we assert the gloo gateway is present. If we remove this assertion, then if a setup file is missing a gateway or something is rejecting the gateway, the test will continue and fail on the longer curl assertion. If we assert here, it fails earlier.